ui/components/src/TimeSeriesTooltip/LineChartTooltip.tsx

```diff
@@ -59,7 +59,7 @@ export const LineChartTooltip = memo(function LineChartTooltip({
   const [showAllSeries, setShowAllSeries] = useState(false);
   const mousePos = useMousePosition();
   const { height, width, ref: tooltipRef } = useResizeObserver();
-  const transform = useRef('');
+  const transform = useRef<string | undefined>();
```

Comment:
Fixed these since type errors were thrown, but LineChartTooltip can also be removed now, as well as legacyCheckforNearbySeries and [getLineSeries](https://github.com/perses/perses/blob/8cf6c9af4ce7d17523269f28fbecc41258dd14f3/ui/panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts#L71) (see [other comment](https://github.com/perses/perses/pull/1750/files#r1482388936))